### ISSUE TITLE
Add an option to remove local source map after upload

### ISSUE THREAD

**Issue by joshuap** [honeybadger-io/honeybadger-js#879]
Package Name: @honeybadger-io/webpack

**Comment by BethanyBerkowitz** [comment]
I have a [WIP PR for this](https://github.com/honeybadger-io/honeybadger-js/pull/973), but I have some concerns.

I tried two approaches: 
1. Remove the sourcemap file right after upload 
2. Remove all the sourcemap files after build, ie in Webpack's `done` hook.  

In both cases, I have the same error when used in a NextJS app and deploying to Vercel:
```
ENOENT: no such file or directory, lstat '/vercel/path0/.next/server/webpack-runtime.js.map'
```
I've also tried going into the js files and removing the reference to the sourcemaps in addition to removing the sourcemap files themselves, like this: 
```
const jsFileContent = await fs.readFile(path, 'utf8')
await fs.writeFile(path, jsFileContent.replace(/\/\/# sourceMappingURL=\S+/g, ''))
```
but the error remains the same. 

My concern here is that it seems like Vercel and/or NextJS is expecting the results of the webpack build to look a certain way, and we are messing with that. I could continue trying to fix this this in a Vercel setup, but my concern is that each platform and build/deploy setup could require a different workaround if we go ahead and just remove files that Webpack created (and that other tools may therefore expect to exist). 

Any thoughts on this? @subzero10 @joshuap ?

**Comment by subzero10** [comment]
@BethanyBerkowitz Have you confirmed that source map files are created in the cases you tested above?
I know that sometimes webpack, when in development mode, won't actually create the files but report that they were created. Have you confirmed that you get the ENOENT error when building in production mode?

**Comment by BethanyBerkowitz** [comment]
> @BethanyBerkowitz Have you confirmed that source map files are created in the cases you tested above? I know that sometimes webpack, when in development mode, won't actually create the files but report that they were created. Have you confirmed that you get the ENOENT error when building in production mode?

<img width="1006" alt="Screen Shot 2022-12-07 at 9 41 12 AM" src="https://user-images.githubusercontent.com/16493493/207076175-f2257ad3-17bf-4e16-a713-db080861c9bd.png">

Yes, the files are successfully uploaded to the HB API (see screenshot). And then they are successfully removed (by the code in my PR). If I deploy without the `removeSourcemaps` flag set, it deploys with no issues.

**Comment by subzero10** [comment]
> In both cases, I have the same error when used in a NextJS app and deploying to Vercel:
> 
> ```
> ENOENT: no such file or directory, lstat '/vercel/path0/.next/server/webpack-runtime.js.map'
> ```

Is it possible that these are next.js internal files?

I'm out of ideas, Next.js has given me headaches more than a few times already 😵 
We could always ignore this flag for Next.js projects 🤷

**Comment by BethanyBerkowitz** [comment]
After working on this for probably too long, I'm not convinced that it's even a good idea. There's a lot we don't know about our users' build and deploy setup, and deleting source map files that webpack generates may have unintended consequences, including cryptic and annoying errors like the one I've been seeing with NextJS. Sentry simply declared this feature out of scope for their webpack plugin, possibly for similar reasons. 

It's one thing if a user has to do some finagling to remove their own sourcemaps, but imo it's worse if they need to do that finagling to make a feature in our plugin work for them. It should just work, and I'm not convinced it will. 

I'm inclined to close this and call this feature out of scope, but would like to hear thoughts from @joshuap

**Comment by subzero10** [comment]
No objection from my side.

And to share my 2 cents:
- I think I came across this only once or twice (having to manually delete js maps), and it's been a long-long time.
- The correct approach would be to have the "build" process part of a CI/CD integration. Or at least it shouldn't be done on a local environment. If that is the case, any generated source maps are lost after the pipeline is complete.
- So, yes, I agree that this feature is unnecessary/redundant or YAGNI for today's deployment mechanisms.

**Comment by joshuap** [comment]
> No objection from my side.
> 
> And to share my 2 cents:
> 
> * I think I came across this only once or twice (having to manually delete js maps), and it's been a long-long time.
> * The correct approach would be to have the "build" process part of a CI/CD integration. Or at least it shouldn't be done on a local environment. If that is the case, any generated source maps are lost after the pipeline is complete.
> * So, yes, I agree that this feature is unnecessary/redundant or YAGNI for today's deployment mechanisms.

I agree. There's nothing preventing the user from adding a manual step in their build pipeline to delete the files they don't want to deploy, and if this isn't an easily reproducible scenario, I think that makes more sense.

Thanks for the work/research on this @BethanyBerkowitz!
